byron/chain/executable-spec/src/Cardano/Spec/Chain/STS/Rule/Chain.hs

```diff
@@ -77,6 +77,15 @@ instance STS CHAIN where
     | LedgerUTxOFailure (PredicateFailure UTXOWS)
     deriving (Eq, Show)
 
+  eqPF (EpochFailure x) (EpochFailure y) = x `eqPF` y
```

Comment:
Here the pattern seems to be: if you have a constructor that wraps another predicate failure, then recurse (?), otherwise compare the constructors only. Is that correct?

Reply:
If so we might want to make an issue about deriving `eqPF` using generics or using some other `syb` traversal, so that this becomes easier to maintain.

Reply:
Yes, that's the pattern. I haven't worked enough with Generics to know how to determine whether to recur or not (i.e. determine whether a field was a `PredicateFailure sub-sts` or not, but if that is possible it would be great.